### ISSUE TITLE
why do you want to leave the download task to the external browser？

### ISSUE THREAD

**Issue by aqianglala** [anthonycr/Lightning-Browser#593]
Hello, why do you want to leave the download task to the external browser? It's like onDownloadStart will never be called, and I don't think it's necessary to add ACCEPTED_URI_SCHEMA, and I'll be waiting for your answer

**Comment by anthonycr** [comment]
I'm using the system download manager because it was simpler than writing my own download manager at the time that code was written. Is there an issue?

**Comment by jmliz** [comment]
@anthonycr yes there is, when you are using anonymizing tools. if you use the system file manager the download will not be routed through Tor. I've already opened an issue at #590

**Comment by DF1E** [comment]
Maybe a DownloadManager like this will help us to improve the security: https://github.com/tonyofrancis/Fetch

**Comment by rancidfrog** [comment]
Also: https://github.com/ssseasonnn/RxDownload

> A multi-threaded download tool written with RxJava and Kotlin
